src/main/scala/mesosphere/marathon/upgrade/ReadinessBehavior.scala

```diff
@@ -79,26 +82,44 @@ trait ReadinessBehavior { this: Actor with ActorLogging =>
     * The #taskIsReady function is called, when the task is ready according to the app definition.
     */
   //scalastyle:off cyclomatic.complexity
-  def readinessBehavior: Receive = {
-    def taskIsRunning(taskFn: Task.Id => Unit): Receive = {
-      case MesosStatusUpdateEvent(slaveId, taskId, "TASK_RUNNING", _, `appId`, _, _, _, `versionString`, _, _) =>
-        taskFn(taskId)
-    }
-
-    def taskIsHealthy(taskFn: Task.Id => Unit): Receive = {
-      case HealthStatusChanged(`appId`, taskId, `version`, true, _, _) if !healthy(taskId) => taskFn(taskId)
+  val readinessBehavior: Receive = {
+
+    def taskRunBehavior: Receive = {
+      def runningTaskWithoutReady(taskId: Task.Id): Unit = {
+        log.debug(s"Started task is ready: $taskId")
+        healthy += taskId
+        ready += taskId
+        taskStatusChanged(taskId)
+      }
+      def runningTaskWithReady(taskId: Task.Id): Unit = {
```

Comment:
After reading line 102, I think a better name would be `markAsHealthyAndInitiateReadinessCheck`... although it is a very long name...